frontend/simple/views/Join.vue

```diff
@@ -1,122 +1,121 @@
-              <div class="panel-block">
-                {{contract.founderUsername}}
-              </div>
-              <div class="panel-block">
-                <div><strong><i18n>Group Info</i18n></strong></div>
-              </div>
-              <div class="panel-block">
-                <span>{{contract.groupName}} <i18n>is a group that was founded</i18n> {{formatDate(contract.creationDate)}}. <i18n>The Group currently has</i18n> {{contract.members.length}} <i18n>active members with</i18n> # <i18n>financial contributors</i18n>.</span>
-              </div>
-              <div class="panel-block">
-                <div><strong><i18n>Percentage of members are required to change the rules</i18n></strong></div>
-              </div>
-              <div class="panel-block">
-                {{contract.changeThreshold | toPercent}}
-              </div>
-              <div class="panel-block">
-                <strong><i18n>Open to New Members?</i18n></strong>
-              </div>
-              <div class="panel-block" v-show="openMembership"><i18n>Yes</i18n></div>
-              <div class="panel-block" v-show="!openMembership"><i18n>No</i18n></div>
-              <div class="panel-block">
-                <strong><i18n>Percentage of members are required to approve a new members</i18n></strong>
-              </div>
-              <div class="panel-block">
-                {{contract.memberApprovalThreshold | toPercent}}
-              </div>
-              <div class="panel-block">
-                <strong><i18n>Percentage of members are required to remove a member</i18n></strong>
-              </div>
-              <div class="panel-block">
-                {{contract.memberRemovalThreshold | toPercent}}
-              </div>
-              <div class="panel-block">
-                <strong><i18n>Contribution Transparency</i18n></strong>
-              </div>
-              <div class="panel-block">
-                {{contract.contributionPrivacy}}
-              </div>
-            </div>
-          </div>
-          <div class="column is-one-quarter">
-            <table class="table is-bordered is-striped is-narrow is-fullwidth">
-            <thead>
-            <tr>
-              <th><i18n>Group Members</i18n></th>
-            </tr>
-            </thead>
-            <tbody>
-            <tr v-for="member in contract.members" class="member">
-              <td>
-                <div class="media">
-                  <div class="media-left">
-                    <p class="image is-64x64">
-                      <img :src="member.attributes.picture">
-                    </p>
-                  </div>
-                  <div class="media-content">
-                    <strong>{{member.attributes.name}}</strong>
-                  </div>
-                </div>
-              </td>
-            </tr>
-            </tbody>
-          </table>
-          </div>
-        </div>
+  <main class="gi-join-grid" v-if="contract.groupName">
```

Comment:
> @greg - What have you noticed with <i18n>? Can you describe in more detail?

This was something hard to notice unless you have a sharpen eye. I used DevTools performance to see the page loading frame by frame, and the issue was not i18 at all.

The content related to `contract` isn't visible until it gets loaded (because it's async). During that loading time the [user sees this layout](https://user-images.githubusercontent.com/14869087/40577456-c19f4dcc-60fd-11e8-8626-4d698f8156ae.png).

To prevent that from happening I've added this validation `v-if="contract.groupName"`. If the the loading is fast enough, the user will just see a blank page being populated with the content at once. If it takes more than `500ms` the user will see a loading message _"Hang in there, 'Magic is happening..."_

I thought this can be reusable in another views / async content, so I've isolated that Loading logic into a `Loading` component.

Reply:
But now the test fails, in this exact step "Should Receive Message and Invite", I believe it's because of this if. The test should somehow wait for the page to finish loading. How do I do it?

Reply:
the way our Nightmare tests work is that we have `.wait` statements that wait for async stuff to happen by waiting for certain elements to arrive, and then we have `.click`s and other interactions to interact with the appearing elements. so probably what you need there is another `.wait` statement at the right place to stop and wait for the `Loading` component to disappear and our tested component to appear. I can take a look at it at my next review turtle step to find where that `.wait` goes, but you can probably find it sooner @sandrina-p :)